.circleci/config.yml

```diff
@@ -183,6 +183,19 @@ jobs:
               eval "$(t/test_postgresql | grep TEST_PG=)" && PERL5LIB=lib prove -v $openqa_unstable_tests || echo FAIL
             fi
 
+  testapi:
+    docker:
+      - <<: *base
+    steps:
+      - checkout
+      - run: *chown_hack_for_cache
+      - restore_cache: *restore_cache
+      - run: *check_cache
+      - run: *install_cached_packages
+      - run: 
+          command: |
+            eval "$(t/test_postgresql | grep TEST_PG=)" && PERL5LIB=lib prove -v t/api/*.t
```

Comment:
I know it's the same as in the other rules but can you explain

1. why does it need so much verbose code when for travis we could get by with a single line?
2. We should use the rules from the Makefile to ensure that people have a single way to call the same commands manually as used in the CI. Why did you decide against that?

Reply:
I also agree that these should be Makefile targets. And this eval-grep construct looks just unnecessarily complicated indeed. You can simply pass a pre-defined path to `t/test_postgresql` and just set the environment variable accordingly.

Reply:
Travis doesn't do it in one line - it needs very complex entrypoint->start->run construct which is hard to keep in memory, troubleshoot and reuse.
E.g. @kalikiana was looking happy to see that there is one-liner which can be reused in any environment instead of complex pre-setup. And @coolo also did agree that setup in travis is way too complicated.
And having pre-defined DB path also brings some disadvantages, splitting setup and run phases and potentially causing issues when leftovers from previous run may interact with next one. (or theoretical parallel testing disadvantages).
But it is opensource after all - you may block others work and demand that everyone should implement what you like or you can suggest what others will have to agree and live with.
(I don't mind if targets identical to travis will be used in circleci e.g. after implementing code coverage, which will enforce something more complex than flat `prove -v`).

Reply:
Regarding 2. - I am in favor of having Makefile rules, just I don't have useful one at the moment, because I agree that `eval oneliner' is not what can be very useful in Makefile

Reply:
> Travis doesn't do it in one line - it needs very complex entrypoint->start->run construct which is hard to keep in memory, troubleshoot and reuse.

I agree that things which were needlessly complicated before should not be taken over and instead be simplified. But as I said this eval-grep construct is also not the most simplest approach and having these as plain Makefile targets possibly reducing redundancy is in my opinion better because then the commands are easier to use.

> But it is opensource after all - you may block others work

I personally don't see this as a blocker.

> And having pre-defined DB path also brings some disadvantages, splitting setup and run phases and potentially causing issues when leftovers from previous run may interact with next one.

I wouldn't do that either. I'd just prefer having these one-liners in a Makefile without the eval-grep construct.

Reply:
> I wouldn't do that either. I'd just prefer having these one-liners in a Makefile without the eval-grep construct.

I don't disagree, but to me they are too ugly to go into make file. (Feel free to add it :) ) 
I see the task as 'start database + run test'. Everyone must be fee to start database where they prefer and pass location further; `run test` should be simple and flexible at the same time to be applicable group of tests or for individual test.